system-addon/content-src/components/ContextMenu/ContextMenu.jsx

```diff
@@ -0,0 +1,42 @@
+const React = require("react");
+
+const ContextMenu = React.createClass({
+  componentWillMount() {
+    this.hideContext = () => {
+      this.props.onUpdate(false);
+    };
+  },
+  componentDidUpdate(prevProps) {
+    if (this.props.visible && !prevProps.visible) {
+      setTimeout(() => {
+        window.addEventListener("click", this.hideContext);
+      }, 0);
+    }
+    if (!this.props.visible && prevProps.visible) {
+      window.removeEventListener("click", this.hideContext);
```

Comment:
This needs to be done in `componentDidUnmount` to prevent possible leakages as well